pkg/nfd-master/nfd-master.go

```diff
@@ -277,13 +287,67 @@ func (m *nfdMaster) Run() error {
 			if m.nfdController != nil && m.args.EnableNodeFeatureApi {
 				m.nfdController.updateAllNodesChan <- struct{}{}
 			}
+			// Restart the node updater pool
+			m.stopNodeUpdaterPool()
+			go m.runNodeUpdaterPool()
+
 		case <-m.stop:
 			klog.Infof("shutting down nfd-master")
 			return nil
 		}
 	}
 }
 
+func (m *nfdMaster) processNodeLabelRequest() bool {
```

Comment:
Safer to use local variable as we operate in different  goroutines and the queue is our "sync channel"

```suggestion
func (m *nfdMaster) processNodeLabelRequest(queue workqueue.RateLimitingInterface) bool {
```